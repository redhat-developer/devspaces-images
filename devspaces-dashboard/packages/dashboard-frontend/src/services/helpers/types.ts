/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { AlertVariant } from '@patternfly/react-core';
import * as React from 'react';
import { Devfile } from '../workspace-adapter';

export interface AlertItem {
  key: string;
  title: string;
  variant: AlertVariant;
  children?: React.ReactNode;
}

export interface FactoryResolver {
  v: string;
  source?: string;
  devfile: Devfile;
  location?: string;
  scm_info?: FactoryResolverScmInfo;
  links: api.che.core.rest.Link[];
}

export type FactoryResolverScmInfo = {
  clone_url: string;
  scm_provider: string;
  branch?: string;
};

export type DevfileV2ProjectSource = {
  name: string;
  git: {
    remotes: { origin: string };
    checkoutFrom?: { revision: string };
  };
};

export type DeprecatedWorkspaceStatus = 'Deprecated';

export enum WorkspaceStatus {
  RUNNING = 'RUNNING',
  STOPPING = 'STOPPING',
  STOPPED = 'STOPPED',
  STARTING = 'STARTING',
  ERROR = 'ERROR',
}

export enum DevWorkspaceStatus {
  FAILED = 'Failed',
  FAILING = 'Failing',
  STARTING = 'Starting',
  TERMINATING = 'Terminating',
  RUNNING = 'Running',
  STOPPED = 'Stopped',
  STOPPING = 'Stopping',
}

export type CreateWorkspaceTab = 'quick-add' | 'custom-workspace';

export enum IdeLoaderTab {
  Progress = 0,
  Logs = 1,
}

export enum WorkspaceDetailsTab {
  OVERVIEW = 'Overview',
  DEVFILE = 'Devfile',
  LOGS = 'Logs',
}

export enum WorkspaceAction {
  OPEN_IDE = 'Open',
  START_DEBUG_AND_OPEN_LOGS = 'Open in verbose mode',
  START_IN_BACKGROUND = 'Start in background',
  STOP_WORKSPACE = 'Stop Workspace',
  DELETE_WORKSPACE = 'Delete Workspace',
  ADD_PROJECT = 'Add Project',
  ADD_CUSTOM_WORKSPACE = 'Add Workspace',
  RESTART_WORKSPACE = 'Restart Workspace',
  EDIT_WORKSPACE = 'Edit Workspace',
}

export type UserPreferencesTab = 'container-registries';
