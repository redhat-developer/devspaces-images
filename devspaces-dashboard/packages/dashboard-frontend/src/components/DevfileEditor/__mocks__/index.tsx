/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React, { ChangeEvent, ClipboardEvent } from 'react';
import { safeDump } from 'js-yaml';
import devfileApi from '../../../services/devfileApi';
import stringify from '../../../services/helpers/editor';

type Props = {
  devfile: che.WorkspaceDevfile | devfileApi.Devfile;
  decorationPattern?: string;
  onChange: (newValue: string, isValid: boolean) => void;
  isReadonly?: boolean;
};
type State = {
  errorMessage: string;
};

export default class DevfileEditor extends React.PureComponent<Props, State> {
  private editor: HTMLTextAreaElement;

  constructor(props: Props) {
    super(props);

    this.state = {
      errorMessage: '',
    };
  }

  public updateContent(devfile: che.WorkspaceDevfile | devfileApi.Devfile): void {
    if (!this.editor) {
      return;
    }
    this.editor.value = stringify(devfile);
  }

  public componentDidMount(): void {
    const element = document.querySelector('#devfile-editor');
    if (element) {
      const value = stringify(this.props.devfile);
      this.editor = element as HTMLTextAreaElement;
      this.editor.value = value;
    }
  }

  public render(): React.ReactElement {
    const { devfile } = this.props;
    const content = safeDump(devfile);

    const onChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
      const newContent = event.currentTarget.value;
      this.onChange(newContent, true);
    };

    const onPaste = (event: ClipboardEvent<HTMLTextAreaElement>) => {
      const newContent = event.clipboardData?.getData('Text') || '';
      this.onChange(newContent, true);
    };

    return (
      <div className="devfile-editor">
        <div className="monaco">
          <textarea
            id="devfile-editor"
            onChange={onChange}
            onPaste={onPaste}
            value={content}
          ></textarea>
        </div>
        <div className="error"></div>
        <a target="_blank" rel="noopener noreferrer">
          Devfile Documentation
        </a>
      </div>
    );
  }

  private onChange(newValue: string, isValid: boolean): void {
    this.props.onChange(newValue, isValid);
  }
}
