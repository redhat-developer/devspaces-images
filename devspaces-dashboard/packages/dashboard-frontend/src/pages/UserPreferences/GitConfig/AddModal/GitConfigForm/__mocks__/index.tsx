/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/AddModal/GitConfigForm';
import * as GitConfigStore from '@/store/GitConfig';

export class GitConfigForm extends React.PureComponent<Props> {
  render() {
    const { onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { name: 'User One' } } as GitConfigStore.GitConfig, false)
          }
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { email: 'user-1@chetest.com', name: 'User One' } }, true)
          }
        />
      </div>
    );
  }
}
