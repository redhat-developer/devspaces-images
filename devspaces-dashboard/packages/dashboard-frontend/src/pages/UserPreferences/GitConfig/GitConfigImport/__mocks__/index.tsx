/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/GitConfigImport';

export class GitConfigImport extends React.PureComponent<Props> {
  render() {
    const { content, onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          value={content}
          onClick={() => onChange('[user]\n\tname = User One\n', false)}
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange('[user]\n\tname = User One\n\temail = user-1@chetest.com\n', true)
          }
        />
      </div>
    );
  }
}
