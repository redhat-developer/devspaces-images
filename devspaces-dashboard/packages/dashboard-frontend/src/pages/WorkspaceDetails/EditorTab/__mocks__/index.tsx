/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import common from '@eclipse-che/common';
import React from 'react';
import { Props, State } from '..';
import devfileApi from '../../../../services/devfileApi';
import { Workspace } from '../../../../services/workspace-adapter';

export default class EditorTab extends React.PureComponent<Props, State> {
  cancelChanges: () => void;

  constructor(props: Props) {
    super(props);

    this.state = {
      devfile: {} as devfileApi.Devfile,
      hasChanges: false,
      isDevfileValid: true,
      hasRequestErrors: false,
      currentRequestError: '',
      isExpanded: false,
      showDevfileV2ConfirmationModal: false,
    };

    this.cancelChanges = () => {
      // no-op
    };
  }

  async handleOnSave(workspace: Workspace) {
    try {
      await this.props.onSave(workspace);
    } catch (e) {
      this.setState({
        currentRequestError: common.helpers.errors.getMessage(e),
      });
    }
  }

  render() {
    return (
      <div>
        Fake Editor Tab
        <button onClick={() => this.handleOnSave(this.props.workspace)}>Save</button>
        <button onClick={() => this.props.onDevWorkspaceWarning()}>Warning</button>
        <span data-testid="current-request-error">{this.state.currentRequestError}</span>
      </div>
    );
  }
}
