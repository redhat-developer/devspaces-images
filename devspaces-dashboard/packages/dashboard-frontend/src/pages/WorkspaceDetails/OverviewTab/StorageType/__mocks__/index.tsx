/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export default class StorageTypeFormGroup extends React.PureComponent<Props, State> {
  render() {
    return <div>Fake Storage Type Form</div>;
  }
}
