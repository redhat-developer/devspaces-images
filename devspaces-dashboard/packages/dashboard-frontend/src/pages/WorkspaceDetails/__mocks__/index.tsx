/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export const mockShowAlert = jest.fn();
export class WorkspaceDetails extends React.PureComponent<Props, State> {
  render() {
    if (!this.props.workspace) {
      return <div>Not found</div>;
    }
    const workspace = this.props.workspace;
    return (
      <div className="WorkspaceDetails">
        Workspace Details Page
        <div data-testid="props-is-loading">{this.props.isLoading}</div>
        <div data-testid="props-old-workspace-path">
          {this.props.oldWorkspaceLocation?.pathname}
        </div>
        <div data-testid="props-show-convert-button">
          {this.props.showConvertButton === true ? 'true' : 'false'}
        </div>
        <div data-testid="props-workspace-id">{this.props.workspace.id}</div>
        <div data-testid="props-workspace-name">{this.props.workspace.name}</div>
        <div data-testid="props-workspaces-link">{this.props.workspacesLink}</div>
        <button onClick={() => this.props.onSave(workspace)}>Save</button>
        <button onClick={() => this.props.onConvert(workspace)}>Convert</button>
      </div>
    );
  }
}
