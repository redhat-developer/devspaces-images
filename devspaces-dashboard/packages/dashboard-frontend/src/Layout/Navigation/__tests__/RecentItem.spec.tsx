/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import NavigationRecentItem from '../RecentItem';
import { NavigationRecentItemObject } from '..';
import { createHashHistory } from 'history';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { WorkspaceStatus } from '../../../services/helpers/types';
import { FakeStoreBuilder } from '../../../store/__mocks__/storeBuilder';

jest.mock('../../../components/Workspace/Indicator', () => {
  return function DummyWorkspaceIndicator(): React.ReactElement {
    return <div>Dummy Workspace Indicator</div>;
  };
});

describe('Navigation Item', () => {
  const item: NavigationRecentItemObject = {
    status: WorkspaceStatus.STOPPED,
    label: 'workspace',
    to: '/namespace/workspace',
    isDevWorkspace: false,
    workspaceId: 'test-wrks-id',
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render navigation item for STOPPED workspace correctly', () => {
    const status = WorkspaceStatus.STOPPED;
    const element = buildElement(Object.assign({}, item, { status }), '', true);

    expect(renderer.create(element).toJSON()).toMatchSnapshot();
  });

  it('should render navigation item for STARTING workspace correctly', () => {
    const status = WorkspaceStatus.STARTING;
    const element = buildElement(Object.assign({}, item, { status }), '', true);

    expect(renderer.create(element).toJSON()).toMatchSnapshot();
  });

  it('should render navigation item for RUNNING workspace correctly', () => {
    const status = WorkspaceStatus.RUNNING;
    const element = buildElement(Object.assign({}, item, { status }), '', true);

    expect(renderer.create(element).toJSON()).toMatchSnapshot();
  });

  it('should render navigation item for STOPPING workspace correctly', () => {
    const status = WorkspaceStatus.STOPPING;
    const element = buildElement(Object.assign({}, item, { status }), '', true);

    expect(renderer.create(element).toJSON()).toMatchSnapshot();
  });

  it('should render navigation item for ERROR workspace correctly', () => {
    const status = WorkspaceStatus.ERROR;
    const element = buildElement(Object.assign({}, item, { status }), '', true);

    expect(renderer.create(element).toJSON()).toMatchSnapshot();
  });

  it('should have correct label', () => {
    render(buildElement(item));

    const link = screen.getByTestId(item.to);
    expect(link).toHaveTextContent('workspace');
  });

  it('should have workspace status icon', () => {
    render(buildElement(item));
    const workspaceStatusIndicator = screen.getByText('Dummy Workspace Indicator');
    expect(workspaceStatusIndicator).toBeDefined();
  });

  describe('activation', () => {
    it('should render not active navigation item', () => {
      render(buildElement(item));

      const link = screen.getByTestId(item.to);
      expect(link).not.toHaveAttribute('aria-current');
    });

    it('should render active navigation item', () => {
      render(buildElement(item, '/namespace/workspace'));

      const link = screen.getByTestId(item.to);
      expect(link).toHaveAttribute('aria-current');
    });

    it('should activate navigation item on props change', () => {
      const { rerender } = render(buildElement(item));

      rerender(buildElement(item, '/namespace/workspace'));

      const link = screen.getByTestId(item.to);
      expect(link).toHaveAttribute('aria-current');
    });
  });
});

function buildElement(
  item: NavigationRecentItemObject,
  activeItem = '',
  isDefaultExpanded = false,
): JSX.Element {
  const store = new FakeStoreBuilder().build();
  const history = createHashHistory();
  return (
    <Provider store={store}>
      <MemoryRouter>
        <NavigationRecentItem
          isDefaultExpanded={isDefaultExpanded}
          item={item}
          activePath={activeItem}
          history={history}
        />
      </MemoryRouter>
    </Provider>
  );
}
