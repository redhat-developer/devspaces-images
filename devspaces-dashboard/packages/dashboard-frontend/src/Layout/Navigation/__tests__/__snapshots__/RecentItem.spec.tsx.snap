// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Item should render navigation item for ERROR workspace correctly 1`] = `
<li
  className="pf-c-nav__item"
  data-ouia-component-id="OUIA-Generated-NavItem-5"
  data-ouia-component-type="PF4/NavItem"
  data-ouia-safe={true}
>
  <a
    aria-current={null}
    className="pf-c-nav__link"
    data-testid="/namespace/workspace"
    onClick={[Function]}
    tabIndex={null}
  >
    <span
      data-testid="recent-workspace-item"
    >
      <div>
        Dummy Workspace Indicator
      </div>
      workspace
    </span>
    <div
      className="pf-c-dropdown pf-m-align-right pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Dropdown-5"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe={true}
      onClick={[Function]}
    >
      <button
        aria-expanded={true}
        aria-haspopup={true}
        aria-label="Actions"
        className="pf-c-dropdown__toggle pf-m-plain"
        disabled={false}
        id="pf-dropdown-toggle-id-4"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-4"
        autoFocus={false}
        className="pf-c-dropdown__menu pf-m-align-right"
        hidden={false}
        role="menu"
      >
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-11"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Stop Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-12"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Restart Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-13"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Edit Workspace
            </div>
          </a>
        </li>
      </ul>
    </div>
  </a>
</li>
`;

exports[`Navigation Item should render navigation item for RUNNING workspace correctly 1`] = `
<li
  className="pf-c-nav__item"
  data-ouia-component-id="OUIA-Generated-NavItem-3"
  data-ouia-component-type="PF4/NavItem"
  data-ouia-safe={true}
>
  <a
    aria-current={null}
    className="pf-c-nav__link"
    data-testid="/namespace/workspace"
    onClick={[Function]}
    tabIndex={null}
  >
    <span
      data-testid="recent-workspace-item"
    >
      <div>
        Dummy Workspace Indicator
      </div>
      workspace
    </span>
    <div
      className="pf-c-dropdown pf-m-align-right pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Dropdown-3"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe={true}
      onClick={[Function]}
    >
      <button
        aria-expanded={true}
        aria-haspopup={true}
        aria-label="Actions"
        className="pf-c-dropdown__toggle pf-m-plain"
        disabled={false}
        id="pf-dropdown-toggle-id-2"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-2"
        autoFocus={false}
        className="pf-c-dropdown__menu pf-m-align-right"
        hidden={false}
        role="menu"
      >
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-7"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Stop Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-8"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Restart Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-9"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Edit Workspace
            </div>
          </a>
        </li>
      </ul>
    </div>
  </a>
</li>
`;

exports[`Navigation Item should render navigation item for STARTING workspace correctly 1`] = `
<li
  className="pf-c-nav__item"
  data-ouia-component-id="OUIA-Generated-NavItem-2"
  data-ouia-component-type="PF4/NavItem"
  data-ouia-safe={true}
>
  <a
    aria-current={null}
    className="pf-c-nav__link"
    data-testid="/namespace/workspace"
    onClick={[Function]}
    tabIndex={null}
  >
    <span
      data-testid="recent-workspace-item"
    >
      <div>
        Dummy Workspace Indicator
      </div>
      workspace
    </span>
    <div
      className="pf-c-dropdown pf-m-align-right pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Dropdown-2"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe={true}
      onClick={[Function]}
    >
      <button
        aria-expanded={true}
        aria-haspopup={true}
        aria-label="Actions"
        className="pf-c-dropdown__toggle pf-m-plain"
        disabled={false}
        id="pf-dropdown-toggle-id-1"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-1"
        autoFocus={false}
        className="pf-c-dropdown__menu pf-m-align-right"
        hidden={false}
        role="menu"
      >
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-4"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Stop Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-5"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Restart Workspace
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-6"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Edit Workspace
            </div>
          </a>
        </li>
      </ul>
    </div>
  </a>
</li>
`;

exports[`Navigation Item should render navigation item for STOPPED workspace correctly 1`] = `
<li
  className="pf-c-nav__item"
  data-ouia-component-id="OUIA-Generated-NavItem-1"
  data-ouia-component-type="PF4/NavItem"
  data-ouia-safe={true}
>
  <a
    aria-current={null}
    className="pf-c-nav__link"
    data-testid="/namespace/workspace"
    onClick={[Function]}
    tabIndex={null}
  >
    <span
      data-testid="recent-workspace-item"
    >
      <div>
        Dummy Workspace Indicator
      </div>
      workspace
    </span>
    <div
      className="pf-c-dropdown pf-m-align-right pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Dropdown-1"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe={true}
      onClick={[Function]}
    >
      <button
        aria-expanded={true}
        aria-haspopup={true}
        aria-label="Actions"
        className="pf-c-dropdown__toggle pf-m-plain"
        disabled={false}
        id="pf-dropdown-toggle-id-0"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-0"
        autoFocus={false}
        className="pf-c-dropdown__menu pf-m-align-right"
        hidden={false}
        role="menu"
      >
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-1"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Open in verbose mode
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-2"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Start in background
            </div>
          </a>
        </li>
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-3"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Edit Workspace
            </div>
          </a>
        </li>
      </ul>
    </div>
  </a>
</li>
`;

exports[`Navigation Item should render navigation item for STOPPING workspace correctly 1`] = `
<li
  className="pf-c-nav__item"
  data-ouia-component-id="OUIA-Generated-NavItem-4"
  data-ouia-component-type="PF4/NavItem"
  data-ouia-safe={true}
>
  <a
    aria-current={null}
    className="pf-c-nav__link"
    data-testid="/namespace/workspace"
    onClick={[Function]}
    tabIndex={null}
  >
    <span
      data-testid="recent-workspace-item"
    >
      <div>
        Dummy Workspace Indicator
      </div>
      workspace
    </span>
    <div
      className="pf-c-dropdown pf-m-align-right pf-m-expanded"
      data-ouia-component-id="OUIA-Generated-Dropdown-4"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe={true}
      onClick={[Function]}
    >
      <button
        aria-expanded={true}
        aria-haspopup={true}
        aria-label="Actions"
        className="pf-c-dropdown__toggle pf-m-plain"
        disabled={false}
        id="pf-dropdown-toggle-id-3"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-3"
        autoFocus={false}
        className="pf-c-dropdown__menu pf-m-align-right"
        hidden={false}
        role="menu"
      >
        <li
          className={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
        >
          <a
            aria-disabled={false}
            className="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-10"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe={true}
            tabIndex={-1}
          >
            <div>
              Edit Workspace
            </div>
          </a>
        </li>
      </ul>
    </div>
  </a>
</li>
`;
